mule-apikit-tools/mule-apikit-scaffolder/src/main/java/org/mule/tools/apikit/output/scopes/APIKitFlowScope.java

```diff
@@ -31,22 +33,29 @@ public APIKitFlowScope(GenerationModel flowEntry) {
   public APIKitFlowScope(GenerationModel flowEntry, boolean isMuleEE) {
     flow = new Element("flow", XMLNS_NAMESPACE.getNamespace());
     flow.setAttribute("name", flowEntry.getFlowName());
-    flow.addContent(generateFlowContent(flowEntry, isMuleEE));
+
+    for (Element element : generateFlowContent(flowEntry, isMuleEE)) {
+      flow.addContent(element);
+    }
   }
 
-  private Element generateFlowContent(GenerationModel flowEntry, boolean isMuleEE) {
+  private List<Element> generateFlowContent(GenerationModel flowEntry, boolean isMuleEE) {
```

Comment:
You shouldn't modify transform nor logger generation. I think a better approach is to add set variables generation before calling those generations. This should simplify the code

Reply:
You're right, I just refactored the code